nautiluszim/scraper.py

```diff
@@ -345,26 +346,56 @@ def test_collection(self):
         for entry in self.json_collection:
             if not entry.get("files"):
                 continue
-            for relative_path in entry["files"]:
-                if relative_path not in all_names:
-                    missing_files.append(relative_path)
+            for file in entry["files"]:
+                path = self.parse_files(file)["path"]
+                if not path.startswith("http") and path not in all_names:
+                    missing_files.append(path)
 
         if missing_files:
             raise ValueError(
                 "File(s) referenced in collection but missing:\n - "
                 + "\n - ".join(missing_files)
             )
 
+    def parse_files(self, file: Union[str, Dict[str, str]]):
+        if isinstance(file, str):
+            return {"path": file, "filename": file}
+        archive_member = file.get("archive-member", None)
+        url = file.get("url", None)
+        path = None
+        filename = None
+        if archive_member is None and url is None:
+            raise ValueError("archive_member and url are both missing")
+        if url:
+            path = url
+            filename = os.path.basename(urlparse(url).path)
+        else:
+            path = archive_member
+            filename = archive_member
+        filename = file.get("filename", filename)
+        return {"path": path, "filename": filename}
+
     def process_collection_entries(self):
         for entry in self.json_collection:
             if not entry.get("files"):
                 continue
 
-            for relative_path in entry["files"]:
-                logger.debug(f"> {relative_path}")
+            for file in entry["files"]:
+                file_entity = self.parse_files(file)
+                path = file_entity["path"]
+                filename = file_entity["filename"]
+                logger.debug(f"> {path}")
+
+                if path.startswith("http"):
+                    save_location = self.output_dir.joinpath(filename).resolve()
```

Comment:
`output_dir` is for final ZIM ; there's `build_dir` for this.

You should use some `tempfile` inside `build_dir` otherwise there's a risk the filename conflicts with another file from the ZIP.